docs/reST/ref/math.rst

```diff
@@ -61,6 +61,38 @@ Multiple coordinates can be set using slices or swizzling
 
    .. ## math.clamp ##
 
+.. function:: lerp
+
+   | :sl:`interpolates between two values by a percentage.`
+   | :sg:`lerp(a, b, percent) -> float`
+
+   Linearly interpolates between ``a`` and ``b`` by ``percent`` using the formula ``a + (b-a) * percent``.
+   
+   If ``percent`` is ``0.5``, ``lerp`` will return the value half-way between ``a``
+   and ``b``. When ``a = 10`` and ``b = 20``, ``lerp(a, b, 0.5)`` will return ``15``.
+
+   ``lerp`` can be used for many things. You could rotate a sprite by a percentage with
+   ``angle = lerp(0, 360, percent)``. You could even scale an enemy's attack value
+   based on the level you're playing:
+
+   ::
+
+      FINAL_LEVEL = 10
+      current_level = 2
+
+      attack = lerp(10, 50, current_level/MAX_LEVEL) # 18
+
+   If you're on level 0, ``attack`` will be ``10``, if you're on level 10,
+   ``attack`` will be ``50``. If you're on level 5, the
+   percentage calculated with ``current_level/MAX_LEVEL`` will be ``0.5``
+   which is 50%, therefore ``attack`` will be ``30``, which is half-way between ``10`` and ``50``.
+
+   Raises a ValueError if ``percent`` is outside the range of ``[0, 1]``.
```

Comment:
So I believe this is still under discussion. I don't have a strong opinion. The other lerp functions in pygame (color and vectors) both raise ValueErrors if the input is outside the range as well, so the consistency is good.

One option that I haven't seen yet is an optional error raise, like `zip`. This could allow people to unlock a "crazy range" lerp if they want, while keeping the error by default, which may help people catch mistaken inputs to the function.